chess2.cpp

```diff
@@ -0,0 +1,882 @@
+            for (;;) {
+                a -= 1;
+                b += 1;
+                if (board[a][b] == -6 || board[a][b] == -2) return true;
+                if (board[a][b] != 0 || a == 0 || b == 7) break;
+            }
+        }
+        a = row;
+        b = col;
+        if (a != 7 && b != 0) {
+            for (;;) {
+                a += 1;
+                b -= 1;
+                if (board[a][b] == -6 || board[a][b] == -2) return true;
+                if (board[a][b] != 0 || a == 7 || b == 0) break;
+            }
+        }
+        a = row;
+        b = col;
+        if (a != 7 && b != 7) {
+            for (;;) {
+                a += 1;
+                b += 1;
+                if (board[a][b] == -6 || board[a][b] == -2) return true;
+                if (board[a][b] != 0 || a == 7 || b == 7) break;
+            }
+        }
+
+        a = row;
+        b = col;
+        if (a != 7) {
+            for (;;) {
+                a += 1;
+                if (board[a][b] == -6 || board[a][b] == -5) return true;
+                if (board[a][b] != 0 || a == 7) break;
+            }
+        }
+        a = row;
+        b = col;
+        if (a != 0) {
+            for (;;) {
+                a -= 1;
+                if (board[a][b] == -6 || board[a][b] == -5) return true;
+                if (board[a][b] != 0 || a == 0) break;
+            }
+        }
+
+        a = row;
+        b = col;
+        if (b != 7) {
+            for (;;) {
+                b += 1;
+                if (board[a][b] == -6 || board[a][b] == -5) return true;
+                if (board[a][b] != 0 || b == 7) break;
+            }
+        }
+        a = row;
+        b = col;
+        if (b != 0) {
+            for (;;) {
+                b -= 1;
+                if (board[a][b] == -6 || board[a][b] == -5) return true;
+                if (board[a][b] != 0 || b == 0) break;
+            }
+        }
+
+        if (row > 0 && col < 6 && board[row - 1][col + 2] == -3)return true;
+        if (row > 1 && col < 7 && board[row - 2][col + 1] == -3)return true;
+        if (row < 7 && col < 6 && board[row + 1][col + 2] == -3)return true;
+        if (row < 6 && col < 7 && board[row + 2][col + 1] == -3)return true;
+        if (row < 6 && col > 0 && board[row + 2][col - 1] == -3)return true;
+        if (row < 7 && col > 1 && board[row + 1][col - 2] == -3)return true;
+        if (row > 1 && col > 0 && board[row - 2][col - 1] == -3)return true;
+        if (row > 0 && col > 1 && board[row - 1][col - 2] == -3)return true;
+
+        if (row != 7 && board[row + 1][col] == -10)return true;
+        if (row != 0 && board[row - 1][col] == -10)return true;
+        if (col != 7 && board[row][col + 1] == -10) return true;
+        if (col != 0 && board[row][col - 1] == -10) return true;
+        if (row != 7 && col != 7 && board[row + 1][col + 1] == -10)return true;
+        if (row != 7 && col != 0 && board[row + 1][col - 1] == -10) return true;
```

Comment:
this can be converted to a generic function where king always go +1  and we have to check for all 8 directions